pkg/kv/kvserver/replica_load.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package kvserver
+
+import "github.com/cockroachdb/cockroach/pkg/util/hlc"
+
+// ReplicaLoad tracks a sliding window of throughput on a replica. By default,
+// there are 6, 5 minute sliding windows.
+type ReplicaLoad struct {
```

Comment:
Do we want a mutex at this level instead of the individual replicaStats levels?

Reply:
I don't think so given they will be called into, within different 3 separate parts of a request/response lifecycle so finer grained locking seems more appropriate here.

Reply:
Are those three parts contending on access here? Especially non of these locks are long-held? Code like the following feels a lot more cumbersome relative if we had a single lock:

```go
			repl.loadStats.requests.mu.Unlock()
 			repl.loadStats.writeKeys.mu.Unlock()
 			repl.loadStats.readKeys.mu.Unlock()
 			repl.loadStats.writeBytes.mu.Unlock()
 			repl.loadStats.readBytes.mu.Unlock()
 			repl.loadStats.batchRequests.mu.Unlock()
```

It's not clear to me that partitioning the locks here by field of access is giving us measurable performance benefit, so I'd err on the side of doing the simpler thing until needing to otherwise (demonstrated by a benchmark).

Reply:
The other reason was to not deprecate the existing qps recording struct leaseholderStats  as of yet. I agree that it might not have any noticeable impact on latency.